File: JavaScript/src/Internals/CalculateTree.js
import { CalculateTreeType } from '../Enums/CalculateTreeType.js';

/**
 * 计算树
 */
export class CalculateTree {
    constructor() {
        this.Type = CalculateTreeType.None;
        this.ErrorMessage = null;
        this.nodes = [];
        this.start = 0;
        this.end = 0;
        this.conditionString = '';
    }
}

// 浏览器支持
if (typeof window !== 'undefined') {
    window.CalculateTree = CalculateTree;
}

File: java/src/main/java/toolgood/algorithm/internals/CalculateTree.java
/**
 * 计算树
 */
package toolgood.algorithm.internals;

import java.util.ArrayList;
import java.util.List;
import toolgood.algorithm.enums.CalculateTreeType;

public final class CalculateTree {
    /**
     * 子节点
     */
    public List<CalculateTree> Nodes;
    /**
     * 开始位置
     */
    public int Start;

    /**
     * 结束位置
     */
    public int End;

    /**
     * 类型
     */
    public CalculateTreeType Type;

    /**
     * 条件
     */
    public String ConditionString;

    /**
     * 出错信息
     */
    public String ErrorMessage;

  
  
}
